### PR TITLE
Two files with switch cases are changed

### PR DESCRIPTION
**Pull request by nyneshreddy**
**PR**
When switch statements have large sets of case clauses, it is usually an attempt to map two sets of data. 
Switch cases can be used when they have single statements. But it is not recommended to write many statements within a single switch case. It is believed that switch cases are bad in Object-Oriented Programming [1]. if-else statement is one of the alternative statement.

The files changed in this pull request are below.

1)inception-pdf-editor/src/main/js/pdfanno/pdfjs/pdf.js
2)inception-pdf-editor/src/main/js/pdfanno/pdfjs/pdf.worker.js

[1] Martin, R. C. (2009). Clean code: a handbook of agile software craftsmanship. Pearson Education.

### CODE REVIEW

File: .sonarcloud.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (munterkalmsteiner) [inline .sonarcloud.properties]: Sonar files should not be included in the project.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+language: java
```
Comment (munterkalmsteiner) [inline .travis.yml]: Why is this needed?

File: inception-pdf-editor/src/main/js/pdfanno/pdfjs/pdf.js [hunk 3]
```diff
@@ -4216,139 +4216,95 @@ var SVGGraphics = (function SVGGraphicsClosure() {
         var fnId = opTree[x].fnId;
         var args = opTree[x].args;
 
-        switch (fnId | 0) {
```
Comment (munterkalmsteiner) [inline inception-pdf-editor/src/main/js/pdfanno/pdfjs/pdf.js:4219]: Why are switch..case statements bad? You need to motivate the change...

**Review by munterkalmsteiner — CHANGES_REQUESTED** [review]

**Review by munterkalmsteiner — CHANGES_REQUESTED** [review]
Ok, I see that you have motivated the change in the issue report. I'd have liked to see also in the PR as motivation, given that the motivation in the issue is rather generic. Also, does the code really improve by replacing switch..case statement with if statements? The issue description does not propose if statements but some other kind of refactoring (map structure).

File: inception-pdf-editor/src/main/js/pdfanno/pdfjs/pdf.js (hunk 3 above)
Comment (nyneshreddy) [reply inception-pdf-editor/src/main/js/pdfanno/pdfjs/pdf.js:4219]: Changes are updated.

**Review by munterkalmsteiner — CHANGES_REQUESTED** [review]
Hm, is the code smell sonar detects the following?
https://rules.sonarsource.com/c/RSPEC-1479

I don't think the correct refactoring is to replace the switch..case construct with a chain of if..else statements. Sonarcube suggests to use a map data structure instead. Did you investigate this?
Here is a good discussion how to implement this properly:
https://codereview.stackexchange.com/questions/42125/refactoring-large-switch-statement

**Comment by nyneshreddy** [comment]
**Since the changes made are in a 3rd party code file this Pull Request is closed**

**Description**
The switch statement, it exists in almost every programing language and javascript is no exception. The switch statement allows for the cleaner organization of code when a large number of evaluations needs to be made. While the switch statement is useful, it doesn’t quite fit in with the rest of our functional code. It’s not Immutable, it can’t be composed with other functions.

**Solution** 
Instead of "Switch" statement a "Data Map" can be created which can be an alternative to switch case and can be reused whenever the data is required which is the principle of object-oriented programming. Moreover, additional data can easily be added to the data map, which makes the code easy to maintain.

**Sample code**
const getDay = switchcase({
    OPS.beginText: this.beginText();,
    OPS.setLeading: this.setLeading(args);,
    OPS.setLeadingMoveText: this.setLeadingMoveText(args[0], args[1]),
    OPS.setFont: this.setFont(args),
    OPS.showText: this.showText(args[0]),
    OPS.showSpacedText: this.showText(args[0]),
    OPS.endText: this.endText(),
	OPS.moveText: this.moveText(args[0], args[1]),
	OPS.setCharSpacing: this.setCharSpacing(args[0]),
	OPS.setWordSpacing: this.setWordSpacing(args[0]),
	OPS.setHScale: this.setHScale(args[0]),
	OPS.setTextMatrix: this.setTextMatrix(args[0], args[1], args[2], args[3], args[4], args[5]),
	OPS.setLineWidth: this.setLineWidth(args[0]),
	OPS.setLineJoin: this.setLineJoin(args[0]),
	OPS.setLineCap: this.setLineCap(args[0]),
	OPS.setMiterLimit: this.setMiterLimit(args[0]),
	OPS.setFillRGBColor: this.setFillRGBColor(args[0], args[1], args[2]),
	OPS.setStrokeRGBColor: this.setStrokeRGBColor(args[0], args[1], args[2]),
	OPS.setDash: this.setDash(args[0], args[1]),
	OPS.setGState: this.setGState(args[0]),
	OPS.fill: this.fill(),
	OPS.eoFill: this.eoFill(),
	OPS.stroke: this.stroke(),
	OPS.fillStroke: this.fillStroke(),
	OPS.eoFillStroke: this.eoFillStroke(),
	OPS.clip: this.clip('nonzero'),
	OPS.eoClip: this.clip('evenodd'),
	OPS.paintSolidColorImageMask: this.paintSolidColorImageMask(),
	OPS.paintJpegXObject: this.paintJpegXObject(args[0], args[1], args[2]),
	OPS.paintImageXObject: this.paintImageXObject(args[0]),
	OPS.paintInlineImageXObject: this.paintInlineImageXObject(args[0]),
	OPS.paintImageMaskXObject: this.paintImageMaskXObject(args[0]),
	OPS.paintFormXObjectBegin: this.paintFormXObjectBegin(args[0], args[1]),
	OPS.paintFormXObjectEnd: this.paintFormXObjectEnd(),
	OPS.closePath: this.closePath(),
	OPS.closeStroke: this.closeStroke(),
	OPS.closeFillStroke: this.closeFillStroke(),
	OPS.nextLine: this.nextLine(),
	OPS.transform: this.transform(args[0], args[1], args[2], args[3], args[4], args[5]),
	OPS.constructPath: this.constructPath(args[0], args[1]),
	OPS.endPath: this.endPath(),
	92: this.group(opTree[x].items),
	default: warn('Unimplemented method '+ fn)
	
  })('Unknown')const getCurrentDay = () =>
  getDay(new Date().getDay())const day = getCurrentDay()
